### PR TITLE
Support for LDAP result type (SAMBA4)

### PR DESCRIPTION
**Pull request by jkolo**

### CODE REVIEW

File: ldap_auth_provider.py [hunk 1]
```diff
@@ -173,8 +173,10 @@ def check_password(self, user_id, password):
                     ]
                 )
 
-                if len(conn.response) == 1:
-                    attrs = conn.response[0]['attributes']
+                response = [r for r in conn.response if r['type'] == 'searchResEntry']
```
Comment (mweinelt) [inline ldap_auth_provider.py]: Don't use single character variables, doesn't make for a good read in my opinion.
```
responses = [response for response in conn.response if response['type'] == 'searchResEntry']

if len(responses) == 1:
```

**Comment by mweinelt** [comment]
So this goes over all responses and filters them for `type == searchResEntry`. 

How does that add support for SAMBA4?

**Comment by mweinelt** [comment]
Can someone with AD verify/explain this patch?

**Comment by jkolo** [comment]
In the past, I had a problem, the answer to the LDAP I had different kinds of type. Today I wanted to repeat the test to see what it was, but at the moment LDAP returns records only with type "searchResEntry"

**Comment by pschrammel** [comment]
This patch solved my simple AD problems on AWS (I think they use SAMBA for this service, it seems to return 4 results)

**Comment by maxidor** [comment]
My 2 cents: I use Samba and the connector works out of the box, so I'm also interested why this is necessary. Maybe it's only a matter of config?

**Comment by mweinelt** [comment]
I'll take some time to look over prs and issues starting next week, maybe @jkolo can rebase the pull request until then?

**Comment by mweinelt** [comment]
I don't see anything harmful in this change and it works for me on slapd, so if it makes someones LDAP work lets merge it.

**Pull request merged**